src/utils/AppRouteProvider.jsx

```diff
@@ -26,27 +60,45 @@ const AppRouteProvider = (props) => {
     ),
   };
 
+  const updateExperimentInfoOnNavigate = (projectUuid, experimentId) => {
+    const lastViewed = moment().toISOString();
+
+    dispatch(switchExperiment(experimentId));
+    dispatch(updateExperiment(experimentId, { lastViewed }));
+    dispatch(updateProject(projectUuid, { lastAnalyzed: lastViewed }));
+  };
+
   const continueNavigation = (nextRoute, hardNavigate) => {
     // Hard navigate, causing the page to refresh and fetch data from server
     if (hardNavigate) window.location.href = nextRoute;
```

Comment:
I think you dont need this `window.location.href` router.Push() is good enough?

Reply:
Or am i missing something

Reply:
It's different. `window.location.href` is used when you want to do "hard navigation", which means you send a request to the server for another page, and the server sends the new page, potentially with new server-side information. Without caching or hydration, this resets all client-side states.

`router.push()` is used for "soft navigation", i.e. the single-page app renders a new page on the client-side only, without requesting a new page to the server. As the page is still the same, this means all the states are kept.

Reply:
Right now no part of the application uses this feature. When navigating to an outside link, we use the `Link` component directly. I can remove this part and make the code simpler.

Reply:
removed as suggested